src/logic/calculate.js

```diff
@@ -0,0 +1,16 @@
+import operate from './operate';
+
+const calculate = (obj, btname) => {
+  const object = obj;
+  const { next, operation } = object;
+  let { total } = object;
+
+  if (btname === '+/-') {
+    total *= -1;
+    object.total = total;
+  } else {
+    object.total = operate(total, next, operation);
+  }
+  return object;
+};
```

Comment:
Note that this is where all the calculator logic should be.
Consider cases when the `btname` is `=` and cases where `btname` is a number. This two cases should be handled differently.